lib/parser/index.js

```diff
@@ -97,10 +105,9 @@ function parse(juttle, options) {
 
     options = processOptions(options, defaultResolver);
 
-    return parseModule({ source: juttle, name: 'main' })
+    return parseModule({ source: juttle, name: 'main', filename: (options.filename ? options.filename : 'main') })
```

Comment:
`options.filename` is always set in `processOptions` so it should be enough to do `filename: options.filename` here.